swatch-producer-aws/src/main/java/com/redhat/swatch/rest/SwatchPskHeaderFilter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.rest;
+
+import io.quarkus.arc.Unremovable;
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+@Slf4j
+// NOTE: without @Unremovable quarkus attempts to optimize this bean out because it's only
+// referenced in application.properties
+@Unremovable
+@ApplicationScoped
+public class SwatchPskHeaderFilter implements ClientRequestFilter {
+
+  @ConfigProperty(name = "SWATCH_SELF_PSK")
+  String psk;
+
+  @Override
+  public void filter(ClientRequestContext requestContext) {
+    requestContext.getHeaders().add("x-rh-swatch-psk", psk);
```

Comment:
is there a check against the psk secret happening somewhere?

Reply:
Should be happening in the internal service, no?

Reply:
Oh yeah I misunderstood what this was doing. Looks good